Objects/unicodeobject.c

```diff
@@ -10673,6 +10673,67 @@ PyUnicode_CompareWithASCIIString(PyObject* uni, const char* str)
     }
 }
 
+int
+PyUnicode_EqualToString(PyObject *unicode, const char *str)
+{
+    assert(_PyUnicode_CHECK(unicode));
+    assert(str);
+    if (PyUnicode_IS_ASCII(unicode)) {
+        size_t len = (size_t)PyUnicode_GET_LENGTH(unicode);
+        return strlen(str) == len &&
```

Comment:
I would prefer to test the length first, to make the code  more readable.

Like:

```
if (strlen(str) == len) {
    return 1;
}
return memcmp(...);
```

Same below.

Reply:
> is more readable than simple return a && b;?

For me, it's easier to reason about a single test per line when I review code.

Keep ``a && b`` if you prefer.

Reply:
The readability problem as I see it, is that your `&&` use has side effects; it is not a pure logic expression.

Reply:
> For me, it's easier to reason about a single test per line when I review code.

Fortunately, every condition here is already on a separate line.

> The readability problem as I see it, is that your `&&` use has side effects; it is not a pure logic expression.

It is how `&&` works in C. There is a lot of code like `arg != NULL and PyDict_Check(arg) && PyDict_GET_SIZE(arg) > count`. I do not think rewriting it in three `if`s with `goto`s can improve readability.